docs/source/reference/pysupported.rst

```diff
@@ -223,6 +223,46 @@ The following functions from the :mod:`operator` module are supported:
 * :func:`operator.truediv`
 * :func:`operator.xor`
 
+.. _pysupported-random:
+
+``random``
+----------
+
+Numba supports top-level functions from the :mod:`random` module, but does
+not allow you to create individual Random instances.  A Mersenne-Twister
+generator is used, with a dedicated internal state.  It is initialized at
+startup with entropy drawn from the operating system.
```

Comment:
Should also put the same note here about the existence of support for the `numpy.random` functions, which do not share state with the Python `random` functions in nopython mode.  Someone might read this document page independent of the Numpy support page.